_aggregations/bucket/auto-interval-date-histogram.md

```diff
@@ -0,0 +1,377 @@
+
+## Example
+
+In the following example, you'll search an index containing blog posts. 
+
+First, create a mapping for this index and specify the `date_posted` field as `date` type:
+
+```json
+PUT blogs
+{
+  "mappings" : {
+    "properties" :  {
+      "date_posted" : {
+        "type" : "date",
+        "format" : "yyyy-MM-dd"
+      }
+    }
+  }
+}
+```
+{% include copy-curl.html %}
+
+Next, index the following documents into the `blogs` index:
+
+```json
+PUT blogs/_doc/1
+{
+  "name": "Semantic search in OpenSearch",
+  "date_posted": "2022-04-17"
+}
+```
+{% include copy-curl.html %}
+
+```json
+PUT blogs/_doc/2
+{
+  "name": "Sparse search in OpenSearch",
+  "date_posted": "2022-05-02"
+}
+```
+{% include copy-curl.html %}
+
+```json
+PUT blogs/_doc/3
+{
+  "name": "Distributed tracing with Data Prepper",
+  "date_posted": "2022-04-25"
+}
+```
+{% include copy-curl.html %}
+
+```json
+PUT blogs/_doc/4
+{
+  "name": "Observability in OpenSearch",
+  "date_posted": "2023-03-23"
+}
+
+```
+{% include copy-curl.html %}
+
+To use the `auto_date_histogram` aggregation, specify the field containing the date or timestamp values. For example, to aggregate blog posts by `date_posted` into two buckets, send the following request:
+
+```json
+GET /blogs/_search
+{
+  "size": 0,
+  "aggs": {
+    "histogram": {
+      "auto_date_histogram": {
+        "field": "date_posted",
+        "buckets": 2
+      }
+    }
+  }
+}
+```
+{% include copy-curl.html %}
+
+The response shows that the blog posts were aggregated into two buckets. The interval was automatically set to 1 year, with all three 2022 blog posts in one bucket and the 2023 blog post in another:
+
+```json
+{
+  "took": 20,
+  "timed_out": false,
+  "_shards": {
+    "total": 1,
+    "successful": 1,
+    "skipped": 0,
+    "failed": 0
+  },
+  "hits": {
+    "total": {
+      "value": 4,
+      "relation": "eq"
+    },
+    "max_score": null,
+    "hits": []
+  },
+  "aggregations": {
+    "histogram": {
+      "buckets": [
+        {
+          "key_as_string": "2022-01-01",
+          "key": 1640995200000,
+          "doc_count": 3
+        },
+        {
+          "key_as_string": "2023-01-01",
+          "key": 1672531200000,
+          "doc_count": 1
+        }
+      ],
+      "interval": "1y"
+    }
+  }
+}
+```
+
+## Returned buckets
+
+Each bucket of the aggregation contains the following information:
+
+```json
+{
+  "key_as_string": "2023-01-01",
+  "key": 1672531200000,
+  "doc_count": 1
+}
+```
+
+In OpenSearch, dates are internally stored as 64-bit integers representing timestamps in milliseconds since the epoch. In the aggregation response, each bucket `key` is returned as such a timestamp. The `key_as_string` value shows the same timestamp, but formatted as a date string based on the [`format`](#date-format) parameter. The `doc_count` field contains the number of documents in the bucket.
```

Comment:
```suggestion
In OpenSearch, dates are internally stored as 64-bit integers representing timestamps in milliseconds since the epoch. In the aggregation response, each bucket `key` is returned as such a timestamp. The `key_as_string` value shows the same timestamp but formatted as a date string based on the [`format`](#date-format) parameter. The `doc_count` field contains the number of documents in the bucket.
```